modules/apps/frontend-js/frontend-js-react-web/test/js/hooks/useInterval.tsx

```diff
@@ -15,41 +15,54 @@
 import {cleanup, fireEvent, render} from '@testing-library/react';
 import React from 'react';
 
-import useInterval from '../../../src/main/resources/META-INF/resources/js/hooks/useInterval.es';
+import useInterval from '../../../src/main/resources/META-INF/resources/js/hooks/useInterval';
 
-const {useEffect, useState} = React;
+const {useState} = React;
 
 const INTERVAL = 100;
 
-const Component = ({callback, onRender, onSchedule}) => {
+type Callback = ReturnType<typeof jest.fn>;
+type Schedule = ReturnType<typeof useInterval>;
+
+const Component = ({
+	callback,
+	onRender,
+	onSchedule,
+}: {
+	callback?: Callback;
+	onRender?: (schedule: Schedule) => void;
+	onSchedule?: (callback: () => void) => void;
+}) => {
 	const schedule = useInterval();
 
-	useEffect(() => {
-		if (onRender) {
-			onRender(schedule);
-		}
-	});
+	if (onRender) {
+		onRender(schedule);
+	}
 
-	const [, forceUpdate] = useState();
+	const [, forceUpdate] = useState(1);
```

Comment:
Was this a bug?

Reply:
I see it's a state... just wondering 😂

Reply:
> Was this a bug?

Nope, just that TS doesn't like it if you pass nothing here, because the type of `useState` is:

```
const useState: {
  <S>(initialState: S | (() => S)): [S, React.Dispatch<React.SetStateAction<S>>];
  <S = undefined>(): [S | undefined, React.Dispatch<React.SetStateAction<S | undefined>>];
}
```

So it infers that the type of `forceUpdate` must be `Dispatch<SetStateAction<undefined>>`. This in turn makes it complain when you do:

```tsx
<button onClick={forceUpdate}>Update</button>
```

with:

```
test/js/hooks/useInterval.tsx:57:12 - error TS2322: Type 'Dispatch<SetStateAction<undefined>>' is not assignable to type 'MouseEventHandler<HTMLButtonElement>'.
  Types of parameters 'value' and 'event' are incompatible.
    Type 'MouseEvent<HTMLButtonElement, MouseEvent>' is not assignable to type 'SetStateAction<undefined>'.
      Type 'MouseEvent<HTMLButtonElement, MouseEvent>' provides no match for the signature '(prevState: undefined): undefined'.

57    <button onClick={forceUpdate}>
```

So, just rejigging things to make it happy. An alternative would have been to leave it as `useState()` and change the `onClick` to this:

```tsx
<button onClick={() => forceUpdate(undefined)}>
```

(You have to pass explicit `undefined` or it will complain about expecting 1 argument but getting 0). This is ugly too, just in a different way. Seeing as this is a test, I didn't want to overthink it and just went with the first approach.